google-cloud-clients/google-cloud-core/src/test/java/com/google/cloud/testing/BaseEmulatorHelperTest.java

```diff
@@ -96,6 +102,40 @@ public void testEmulatorHelper() throws IOException, InterruptedException, Timeo
     EasyMock.verify();
   }
 
+  @Test
+  public void testEmulatorHelperDownloadWithRetries() throws IOException, InterruptedException, TimeoutException {
+    String mockExternalForm = "mockExternalForm";
+    String mockInputStream = "mockInputStream";
+    String mockProtocol = "mockProtocol";
+    String mockFile = "mockFile";
+    String mockCommandText = "mockCommandText";
+
+    MockURLStreamHandler mockURLStreamHandler = EasyMock.createMock(MockURLStreamHandler.class);
+    URLConnection mockURLConnection = EasyMock.mock(URLConnection.class);
+
+    EasyMock.expect(mockURLStreamHandler.toExternalForm(EasyMock.anyObject(URL.class)))
+            .andReturn(mockExternalForm).anyTimes();
+    EasyMock.expect(mockURLConnection.getInputStream())
+            .andReturn(new ByteArrayInputStream(mockInputStream.getBytes())).anyTimes();
+    EasyMock.expect(mockURLStreamHandler.openConnection(EasyMock.anyObject(URL.class)))
+            .andThrow(new EOFException()).times(1);
+    EasyMock.expect(mockURLStreamHandler.openConnection(EasyMock.anyObject(URL.class)))
+            .andReturn(mockURLConnection).times(1);
+    EasyMock.replay(mockURLStreamHandler, mockURLConnection);
+
+    URL url = new URL (mockProtocol, null, 0, mockFile, mockURLStreamHandler);
+    BaseEmulatorHelper.DownloadableEmulatorRunner runner =
+            new BaseEmulatorHelper.DownloadableEmulatorRunner(ImmutableList.of(mockCommandText), url, null);
+
+    File cachedFile = new File(System.getProperty("java.io.tmpdir"), mockExternalForm);
+    cachedFile.delete(); //Clear the cached version so we're always testing the download
```

Comment:
We probably want to make sure that any file is cleaned up after the test completes.